tests/system/test_misc.c

```diff
@@ -338,3 +338,46 @@ int cmd_getuids(void)
 
    return 0;
 }
+
+int cmd_exit_cb(int argc, char **argv)
+{
+   int wstatus;
+   int child_pid;
+   long before_cb;
+   long after_cb;
+
+   do_sysenter_call3(TILCK_CMD_SYSCALL,
+                     TO_PTR(TILCK_CMD_GET_VAR_LONG),
+                     "test_on_exit_cb_counter",
+                     &before_cb);
+
+   DEVSHELL_CMD_ASSERT(before_cb == 3);
+
+   child_pid = fork();
+
+   if (child_pid < 0) {
+      fprintf(stderr, "%s\n", "fork() failed");
+      return 1;
+   }
+
+   if (!child_pid) {
+      do_sysenter_call2(TILCK_CMD_SYSCALL,
+                        TO_PTR(TILCK_CMD_CALL_FUNC_0),
+                        "register_test_on_exit_callback");
+      exit(0);
+   }
+   waitpid(child_pid, &wstatus, 0);
```

Comment:
Sure, I'm interested in doing it in my free time, thanks for the feedback. By the way, about those changes, should I open an issue to keep "track" of it?

Reply:
@0xdeadbad Sure, if you plan to work on that, no problem. It would be a nice task simpler than this one. You can mention in the issue that the handling for the "skipped" case should happen at all levels: kernel self tests, system tests run by the Tilck's devshell and all the 3 test runners (single_test_run, run_interactive_test and run_all_tests).